components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -309,12 +311,20 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         log.debug({ userId: this.userID }, `clientRegion: ${clientHeaderFields.clientRegion}`);
         log.debug({ userId: this.userID }, "initializeClient");
 
-        this.listenForWorkspaceInstanceUpdates();
+        if (this.client && clientMetadata.type === "browser") {
```

Comment:
Why are notifications only relevant for browser clients?
Would be great to have the answer captured in a comment! :orange_heart:

Reply:
> Why are notifications only relevant for browser clients?

Going to add this comment on the final pass. 

But in general, you'd disagree about this change? Those events are only consumed by browser clients, no? 
cc. @akosyakov on this question.

Reply:
I'm not sure what the motivation is. Looking at the other client types ([source](https://github.com/gitpod-io/gitpod/blob/7f82c15f0f9f55d10b1429c7b3b0ff73c0e99a35/components/server/src/websocket/websocket-connection-manager.ts#L60-L67)), I think some others are interested in workspace instance updates as well.
And that makes me wonder if it's worth to introduce this distinction here, based on a field that so far we only used for observability purposes. :shrug:

Reply:
In the end, the biggest problem to solve here is the piling up of listener registrations for go-clients, which is AFAIC primarily the PAPI servers holding client connections 🙈

Reply:
Comment added.

Reply:
all IDE clients, including supervisor/Public API listening for instance updates, they cannot function otherwise properly.

Reply:
> In the end, the biggest problem to solve here is the piling up of listener registrations for go-clients, which is AFAIC primarily the PAPI servers holding client connections

Ok, then I feel I'm lagging context on why we try to limit it here. See Anton's comment.
Maybe we should sync on Monday quickly to get this out the door? :thinking:

Reply:
I see. Didn't check too deeply with all IDE's, but only found no-op handlers for instance updates.
Going to change the PR and only disable prebuild updates to be disabled for non-browser clients.

Reply:
@akosyakov, I reverted the changes to instance updates.

That said, I believe it would make sense to change the way instance updates are forwarded via PAPI server. If that would directly subscribe on redis, we would be in a position where no connection pooling is required anymore in PAPI server.

Reply:
Why don't we observe it for instance updates but only for prebuilds?

Reply:
We reverted the check for browser and proceed with checking for existing clients.